src/main/java/teammates/common/datatransfer/InstructorAttributes.java

```diff
@@ -324,18 +314,9 @@ public boolean hasTutorPrivileges() {
         return privileges.hasTutorPrivileges();
     }
 
-    /**
-     * pre-condition: instructorPrivilegesAsText and privileges should be non-null
-     * @param instructor
-     * @return
-     */
     public boolean isEqualToAnotherInstructor(InstructorAttributes instructor) {
-        if (gson.toJson(this).equals(gson.toJson(instructor))) {
-            return true;
-        } else {
-            return !instructorPrivilegesAsText.equals(instructor.instructorPrivilegesAsText) 
-                   && privileges.equals(instructor.privileges);
-        }
+        JsonParser parser = new JsonParser();
+        return parser.parse(getJsonString()).equals(parser.parse(instructor.getJsonString()));
```

Comment:
Ok, but I still somehow think that it's possible to use `Gson`. Perhaps you can try, and if it does fail, add a comment to explain why `JsonParser` in place of `Gson` is needed.